snapshot/store/groupcache_store.go

```diff
@@ -57,8 +63,26 @@ func MakeGroupcacheStore(underlying Store, cfg *GroupcacheConfig, stat stats.Sta
 			}
 			dest.SetBytes(data)
 			return nil
-		},
-	))
+		}),
+		groupcache.PutterFunc(func(ctx groupcache.Context, bundleName string, data []byte) error {
+			log.Info("New bundle, write and populate cache: ", bundleName)
+			stat.Counter(stats.GroupcacheWriteUnderlyingCounter).Inc(1)
+			defer stat.Latency(stats.GroupcacheWriteUnderlyingLatency_ms).Time().Stop()
+
+			// Get TTL via context type assertion, or set to nil and let underlying use its default
+			ttlc, ok := ctx.(*ttlContext)
```

Comment:
Tests to make sure this is propagated correctly?

Reply:
We could test this in scoot, but it doesn't really belong here. The functionality is tested in the groupcache repo. A test in scoot could also pass as a false positive if e.g. the consistent hash changed.